icefoganalytics/elcc-data-management
# ELCC-57: Split Centre Creation and Update Forms Apart

Relates to:

- https://yg-hpw.atlassian.net/browse/ELCC-57
- https://github.com/icefoganalytics/elcc-data-management/pull/75

# Context

After adding in the various fields for https://yg-hpw.atlassian.net/browse/ELCC-57, the creation dialog became a bit intimidating. As such the creation and update dialogs have been broken apart. The creation dialog is now quite short, and generally similar to the original, pre- new fields version, with the addition of the new required fields.

# Implementation

Show dialogs from query params for better accessibility.

# Screenshots

Showing centre creation
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/5a145d99-1556-4f1c-99bc-2fb57fd57d9c)

# Testing Instructions

1. Boot the app via `dev up`
2. Check that the app complies, and that you can log in at http://localhost:8080.
3. Go to http://localhost:8080/child-care-centres
4. Try to add a center from the button in the top right.
5. Note that the url param updates, and if you refresh the page, the modal re-opens.
6. Make sure that the page redirects after creation.
7. Click on the "pencil" icon of the Child Care Centre Details panel on the left.
8. Note that the url param updates, and if you refresh the page, the modal re-opens.
9. Update some centre details and save the record.
10. Check that the page doesn't break after saving.
